pkg/generate/app/app.go

```diff
@@ -109,32 +111,46 @@ func (r *SourceRef) SuggestName() (string, bool) {
 	if len(r.Name) > 0 {
 		return r.Name, true
 	}
-	return nameFromGitURL(r.URL)
+	if r.URL != nil {
+		return nameFromGitURL(r.URL)
+	}
+	return "", false
 }
 
 // BuildSource returns an OpenShift BuildSource from the SourceRef
 func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
-	return &buildapi.BuildSource{
+	triggers := []buildapi.BuildTriggerPolicy{
+		{
+			Type: buildapi.GitHubWebHookBuildTriggerType,
+			GitHubWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
+		{
+			Type: buildapi.GenericWebHookBuildTriggerType,
+			GenericWebHook: &buildapi.WebHookTrigger{
+				Secret: generateSecret(20),
+			},
+		},
```

Comment:
On Sep 17, 2015, at 8:31 PM, Ben Parees <notifications@github.com> wrote:

In pkg/generate/app/app.go
<https://github.com/openshift/origin/pull/4717#discussion_r39814942>:

>  }
>
>  // BuildSource returns an OpenShift BuildSource from the SourceRef
>  func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTriggerPolicy) {
> -	return &buildapi.BuildSource{
> +	triggers := []buildapi.BuildTriggerPolicy{
> +		{
> +			Type: buildapi.GitHubWebHookBuildTriggerType,
> +			GitHubWebHook: &buildapi.WebHookTrigger{
> +				Secret: generateSecret(20),
> +			},
> +		},
> +		{
> +			Type: buildapi.GenericWebHookBuildTriggerType,
> +			GenericWebHook: &buildapi.WebHookTrigger{
> +				Secret: generateSecret(20),
> +			},
> +		},

Good point. Maybe you don't want it? But we should probably use the
defaulter on a lot of other stuff. We could probably reuse the secret too.

you might not..but the distinction right now is "if you do a source build,
you want it, and if you do a docker build you don't", that seems pretty
arbitrary as it is, might as well be consistent in shoving it down people's
throats.



Agree.  Of course, if we default it, you can't disable it.  That may have
been the reason originally.


(though i'm also not even clear on under which circumstances this method
wouldn't be called by BuildConfig().. when do you ever invoke BuildConfig
but not have source? when defining a custom strategy? does new-app even
support that? or is it for new-build?)



Maybe you could have a custom build that installed a Pom file or something
passed in an env.  I question why you would want to, once we have more
source types defined.


—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4717/files#r39814942>.